src/OpenTelemetry.Exporter.Geneva/Metrics/OtlpProtobuf/OtlpProtobufMetricExporter.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Runtime.InteropServices;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Resources;
+
+namespace OpenTelemetry.Exporter.Geneva;
+
+internal sealed class OtlpProtobufMetricExporter : IDisposable
+{
+    private const int BufferSize = 65360; // the maximum ETW payload (inclusive)
+
+    private readonly byte[] buffer = new byte[BufferSize];
+
+    private readonly OtlpProtobufSerializer otlpProtobufSerializer;
+
+    private readonly Func<Resource> getResource;
+
+    public OtlpProtobufMetricExporter(Func<Resource> getResource)
+    {
+        if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            // Temporary until we add support for user_events.
+            throw new NotSupportedException("Exporting data in protobuf format is not supported on Linux.");
+        }
+
+        this.getResource = getResource;
+        this.otlpProtobufSerializer = new OtlpProtobufSerializer(MetricEtwDataTransport.Instance);
+    }
+
+    public ExportResult Export(in Batch<Metric> batch)
+    {
+        var result = ExportResult.Success;
+
+        int currentPosition = 0;
+
+        try
+        {
+            this.otlpProtobufSerializer.SerializeAndSendMetrics(this.buffer, ref currentPosition, this.getResource(), batch);
```

Comment:
This method should always start writing from the beginning of the buffer. There is no need of `currentPosition` parameter for this method.